scripts/items/pinch_of_stygian_ash.lua

```diff
@@ -0,0 +1,26 @@
+-----------------------------------
+-- ID: 19210
+-- Stygian Ash
+-----------------------------------
+local itemObject = {}
+
+itemObject.onItemEquip = function(target, item)
+    target:addListener('RANGE_STATE_EXIT', 'STYGIAN_ASH_THROWN', function(playerArg, targetArg, action)
```

Comment:
This being in the item looks a bit stinky, wouldn't it make more sense to put range_attack_give/ranged_attack_take triggers in `CCharEntity::OnRangedAttack`; like we have with weaponskills?

```cpp
            this->PAI->EventHandler.triggerListener("WEAPONSKILL_USE", CLuaBaseEntity(this), CLuaBaseEntity(PTargetFound), PSkill->getID(), state.GetSpentTP(), CLuaAction(&action), damage);
            PTarget->PAI->EventHandler.triggerListener("WEAPONSKILL_TAKE", CLuaBaseEntity(PTargetFound), CLuaBaseEntity(this), PSkill->getID(), state.GetSpentTP(), CLuaAction(&action));
```

Reply:
I wouldn't remove the new trigger point though, more triggers isn't a bad thing 👍

Reply:
(if there are no attached listeners, the call is next to free)

Reply:
```cpp
    // calls event from core
    template <class... Args>
    void triggerListener(std::string const& eventname, Args&&... args)
    {
        TracyZoneScoped;
        TracyZoneString(eventname);
        if (eventListeners.count(eventname))
        {
            for (auto&& event : eventListeners.at(eventname))
            {
                auto result = event.lua_func(std::forward<Args&&>(args)...);
                if (!result.valid())
                {
                    sol::error err = result;
                    ShowError("Error in listener event %s: %s", eventname, err.what());
                }
            }
        }
    }

```

Reply:
Interesting so you're saying we should create a ranged_take listener to go alon with the range_state_end listener?

Then, add the ranged take listener to DI? I suppose we could, though we probably wouldn't want it on DI all the time since it's only relevant when he's not engaged? Not impossible to add it on spawn, remove it on engage, and add it on disengage.

Definitely seems simpler in my mind to have it on the item. Also we would need to pass the itemid if the listener on DI would be deciding if he gets claimed or not. (Though I do agree having the listener being removed via a timer because the ammo gets unequipped before the range state ends is a bit wonky)

Reply:
The exit of an entity state is not always perfectly in line with the give/take of what is happening with the state (ranged attack, weaponskill, item use, etc.). I don't think it _hurts_ to keep range_state_end trigger, even if we don't use it. I think it makes more sense for the flow of:
- I (Ixion) install a listener for when I'm hit with a ranged attack. If that ranged attack is from a player with Stygian Ash equipped, do X.

Instead of:
- When a player equips Stygian Ash, install a listener on them so that if they hit something with it it'll have a localvar set. The target can check for this localvar at their convenience. They will do X on their next tick if they have the localvar. 

That seems a lot more steps and a lot less straightforward

Reply:
True. also this code i believe spits out a warning in the luabaseentity constructor for PTarget when a ranged attack is interupted

Reply:
btw I haven't touched this in a while, but 
> I (Ixion) install a listener for when I'm hit with a ranged attack. If that ranged attack is from a player with Stygian Ash equipped, do X.

Note that when the ranged attack finishes, the item is no longer equipped. My original implementation (in my previous PR that got accidentally deleted due to branch deletion) was to create a new AE type that would be re-usable for other arbitrary AE types. Could make it something like `setLocalVar(itemID.."hit")`... but I digress, as it was requested to use a listener.

The listener mechanism in this PR works well and is simple to bypass the issue of single-use ranged items being gone once the hit is registered. I've thought a few times and I just don't know a clean way to do it your suggested way. It would also be a listener that gets called throughout the whole fight. Also, imho it's more obvious where to see how the stygian ash works by looking at the ash lua file